### ISSUE TITLE
ModernBERT training loss is astronomical when using gradient accumulation

### ISSUE THREAD

**Issue by umarbutler** [huggingface/transformers#38982]
### System Info

- `transformers` version: 4.52.4
- Platform: Linux-6.6.87.1-microsoft-standard-WSL2-x86_64-with-glibc2.35
- Python version: 3.12.5
- Huggingface_hub version: 0.33.0
- Safetensors version: 0.5.3
- Accelerate version: 1.8.1
- Accelerate config:    not found
- DeepSpeed version: not installed
- PyTorch version (GPU?): 2.7.1+cu126 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using distributed or parallel set-up in script?: No
- Using GPU in script?: Yes
- GPU type: NVIDIA GeForce RTX 4090

### Who can help?

@SunMarc @ArthurZucker

### Information

- [ ] The official example scripts
- [x] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [x] My own task or dataset (give details below)

### Reproduction

1. Train ModernBERT on MLM with gradient accumulation.

### Expected behavior

Loss values should be the same as they would be without gradient accumulation. Instead, they get multiplied by gradient accumulation steps.
